spring-cloud-dataflow-container-registry/src/main/java/org/springframework/cloud/dataflow/container/registry/ContainerImageRestTemplateFactory.java

```diff
@@ -171,19 +177,25 @@ public void checkServerTrusted(java.security.cert.X509Certificate[] certs, Strin
 		SSLContext sslContext = SSLContext.getInstance("SSL");
 		// Install trust manager to SSL Context.
 		sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+		ConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE);
```

Comment:
It took me a bit to figure out what was new here. My final analysis is that we just have to jump through another hoop to construct the required http client builder. 

I reworked this a bit to abstract out the "get http client builder" which hides the new extra unimportant confusing steps.  Wdyt about replacing this chunk w/ the following?:

```suggestion
		// Create a RestTemplate that uses custom request factory
		return initRestTemplate(
				httpClientBuilder(sslContext),
				withHttpProxy,
				extra);
	}

	private HttpClientBuilder httpClientBuilder(SSLContext sslContext) {
		// Register http/s connection factories
		Lookup<ConnectionSocketFactory> connSocketFactoryLookup = RegistryBuilder.<ConnectionSocketFactory> create()
				.register("https", new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE))
				.register("http", new PlainConnectionSocketFactory())
				.build();
		return HttpClients.custom()
				.setConnectionManager(new BasicHttpClientConnectionManager(connSocketFactoryLookup));
	}
```